// chromogen/build/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Editor = exports.ChromogenZustandObserver = exports.chromogenZustandMiddleware = exports.ChromogenObserver = exports.selectorFamily = exports.atomFamily = exports.selector = exports.atom = void 0;
const api_1 = require("./recoil_generator/src/api/api");
Object.defineProperty(exports, "atom", { enumerable: true, get: function () { return api_1.atom; } });
Object.defineProperty(exports, "selector", { enumerable: true, get: function () { return api_1.selector; } });
Object.defineProperty(exports, "atomFamily", { enumerable: true, get: function () { return api_1.atomFamily; } });
Object.defineProperty(exports, "selectorFamily", { enumerable: true, get: function () { return api_1.selectorFamily; } });
const ChromogenZustandObserver_1 = require("./zustand_generator/src/component/ChromogenZustandObserver");
Object.defineProperty(exports, "ChromogenZustandObserver", { enumerable: true, get: function () { return ChromogenZustandObserver_1.ChromogenZustandObserver; } });
const ChromogenObserver_1 = require("./recoil_generator/src/component/ChromogenObserver");
Object.defineProperty(exports, "ChromogenObserver", { enumerable: true, get: function () { return ChromogenObserver_1.ChromogenObserver; } });
const api_2 = require("./zustand_generator/src/api/api");
Object.defineProperty(exports, "chromogenZustandMiddleware", { enumerable: true, get: function () { return api_2.chromogenZustandMiddleware; } });
const Editor_1 = __importDefault(require("./zustand_generator/src/component/Editor"));
exports.Editor = Editor_1.default;
