// chromogen/build/zustand_generator/src/component/Numbers.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = __importDefault(require("react"));
const listStyle = {
    paddingBlock: '28px',
    display: 'flex',
    flexDirection: 'column',
    alignContent: 'flex-end',
    textAlign: 'end',
    background: '#1c1c1c',
    paddingInline: '16px',
    border: '1px solid #1c1c1c',
    height: 'auto',
    width: '60px',
};
const numberStyle = {
    fontSize: 12,
    color: '#747478',
    fontFamily: 'ui-monospace, SFMono-Regular, SF Mono, Consolas, Liberation Mono, Menlo,monospace',
};
const unique = (val) => (react_1.default.createElement("p", { key: 'number' + val, style: numberStyle }, val));
const numerous = (num = 1000) => {
    let pointer = 1;
    let allNumbers = [];
    while (pointer < num) {
        allNumbers = [...allNumbers, unique(pointer)];
        pointer++;
    }
    console.log(allNumbers);
    return allNumbers;
};
const NumberList = ({ number }) => react_1.default.createElement("div", { style: listStyle }, numerous(number));
exports.default = NumberList;
